cmd/manager/main.go

```diff
@@ -221,7 +222,8 @@ func newRootCommand() *cobra.Command {
 
 	cmd.PersistentFlags().StringVar(&opts.LogLevel, "log-level", defaultLogLevel, "Log level (debug,info,warn,error,fatal)")
 	cmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
-	cmd.PersistentFlags().StringSliceVar(&opts.Controllers, "controllers", opts.Controllers, "Comma-separated list of controllers to run")
+	controllersAsStringSliceVar := getControllersAsStringSliceVar(opts.Controllers)
+	cmd.PersistentFlags().StringSliceVar(&controllersAsStringSliceVar, "controllers", controllersAsStringSliceVar, "Comma-separated list of controllers to run")
```

Comment:
The `controllersAsStringSliceVar` variable is set by the command-line flag, but it is never used.

Reply:
Should it be removed?

Reply:
The `--controllers` flag needs to be set somewhere. The rest of the code assumes that it will be set in `opts.Controllers`. The easiest thing is probably to keep `opts.Controllers` as a `[]string` rather than changing it to a `[]ControllerName`. Then, the `--controllers` flag can still drive that variable directly. Otherwise, there will need to be some other code to take the slice filled out by the `--controllers` flag and set `opts.Controllers` appropriately.